### PR TITLE
[GraphQL] Add some documentation about queries and mutations

### PR DESCRIPTION
**Pull request by alanpoulain**

### CODE REVIEW

File: core/graphql.md [hunk 1]
```diff
@@ -35,6 +35,98 @@ api_platform:
...
+
+If you don't know what a query is yet, the document about them is [here](https://graphql.org/learn/queries/).
+
+For each resource, two queries are available: one for retrieving an item and the other one for the collection.
+For example, if you have a `Book` resource, the queries `book` and `books` can be used.
+
+### Global Object Identifier
+
+When querying an item, you need to pass an identifier as argument. Following the [Relay Global Object Identification Specification](https://facebook.github.io/relay/graphql/objectidentification.htm),
+the identifier needs to be globally unique. That's why in API Platform, you have to use the **IRI** of the resource for this argument.
```
Comment (soyuka) [inline core/graphql.md]:
```suggestion
the identifier needs to be globally unique. In API Platform, this argument will be represented as an [IRI (Internationalized Resource Identifier)](https://www.w3.org/TR/ld-glossary/#internationalized-resource-identifier).
```

WDYT? I'd definitely add the link to the w3.org website.

File: core/graphql.md [hunk 2]
```diff
@@ -35,6 +35,98 @@ api_platform:
...
+
+For example, to query a book having the id 89, you have to do:
```
Comment (soyuka) [inline core/graphql.md]:
```suggestion
For example, to query a book having as identifier `89`, you have to query the following:
```

File: core/graphql.md [hunk 3]
```diff
@@ -35,6 +35,98 @@ api_platform:
...
+
+```graphql
+{
+  book(id: "/books/89") {
+    title
+    isbn
+  }
+}
+```
+
```
Comment (soyuka) [inline core/graphql.md]: > Note that in this example, we're retrieving two fields: `title` and `isbn`.

File: core/graphql.md [hunk 4]
```diff
@@ -35,6 +35,98 @@ api_platform:
...
+        id
+      }
+    }
+  }
+}
+```
+
+A pair of parameters works with the query: `first` and `after`.
+
+More precisely:
```
Comment (soyuka) [inline core/graphql.md]: The two parameters `first` and `after` are necessary to make the paginated query work, more precisely:

File: core/graphql.md [hunk 5]
```diff
@@ -35,6 +35,98 @@ api_platform:
...
+
+* `first` corresponds to the items per page starting from the beginning;
+* `after` corresponds to the `cursor` from which the items are returned.
+
+The current page always has an `endCursor` present in the `pageInfo` field.
+
```
Comment (soyuka) [inline core/graphql.md]:
```suggestion
```

File: core/graphql.md [hunk 6]
```diff
@@ -35,6 +35,98 @@ api_platform:
...
+```graphql
+{
+  offers(first: 10, after: "endCursor") {
+  }
+}
+```
+
+How do you know when you have reached the last page? It is the aim of the property `hasNextPage` in `pageInfo`.
+When it is false, you know it is the last page and moving forward will give you an empty result.
```
Comment (soyuka) [inline core/graphql.md]: When the property `hasNextPage` of the `pageInfo` field is `false`, you've reach the last page. If you move forward, you'll end up having an empty result.

File: core/graphql.md [hunk 7]
```diff
@@ -35,6 +35,98 @@ api_platform:
...
+
+## Mutations
+
+If you don't know what a mutation is yet, the document about them is [here](https://graphql.org/learn/queries/#mutations).
```
Comment (soyuka) [inline core/graphql.md]: a mutation => them ? `it` would be better?

File: core/graphql.md [hunk 8]
```diff
@@ -35,6 +35,98 @@ api_platform:
...
+
+For each resource, three mutations are available: one for creating it (`create`), one for updating it (`update`) and one for deleting it (`delete`).
```
Comment (soyuka) [inline core/graphql.md]: one it repetition could be avoided?

Comment (alanpoulain) [reply core/graphql.md]: I don't know how :confused:

**Pull request merged**